lib/realtime/tenants/check.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Realtime.Tenants.Check do
+  use GenServer
+
+  require Logger
+
+  alias Realtime.Helpers
+
+  def connection_status(tenant_id) do
+    case get_status(tenant_id) do
+      :undefined ->
+        :ok
+        node = Realtime.Nodes.get_node_for_tenant_id(tenant_id)
+
+        case :rpc.call(node, __MODULE__, :set_status, [tenant_id]) do
+          :ok ->
+            {_, res} = get_status(tenant_id)
+            res
+
+          error ->
+            error
+        end
+
+      {_, %{healthy?: true}} ->
+        :ok
+
+      _ ->
+        {:error, :tenant_database_unavailable}
+    end
+  end
+
+  def set_status(tenant_id) do
+    __MODULE__
+    |> Process.whereis()
+    |> Process.send({:set_status, tenant_id}, [])
+
+    set_status_backoff(tenant_id)
+  end
+
+  def get_status(tenant_id) do
+    :syn.lookup(__MODULE__, tenant_id)
+  end
+
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
+  end
+
+  def init(state), do: {:ok, state, {:continue, :setup_syn}}
+
+  def handle_continue(:setup_syn, state) do
+    :ok = :syn.add_node_to_scopes([__MODULE__])
+    {:noreply, state}
+  end
+
+  def handle_info({:set_status, tenant_id}, state) do
+    res = check_tenant_connection(tenant_id)
```

Comment:
And you don't really actually have to do this either.

If this GenServer starts the Postgrex connection then we know it's alive if this GenServer is alive.

And if we register this GenServer with :syn then we know the connection is alive and this info is distributed for us.

Reply:
Also this is going to blow up when you have a bunch of tenants connect at the same time and all of them hit this handle info where this function could take a long time.

Reply:
On this Check.exs Genserver we connect the node to the database and won't stop the connection correct?

Should we keep track of the health status and pid of the postgrex conn in the metadata then?

Reply:
>we connect the node to the database and won't stop the connection correct?

Correct.

>Should we keep track of the health status and pid of the postgrex conn in the metadata then?

You can add metadata to a process in syn but if we register it with syn the fact that it's in syn should mean it's alive and healthy.